File: LessonLockerJPA/src/main/java/com/skilldistillery/lessonlocker/entities/QuizAnswer.java
package com.skilldistillery.lessonlocker.entities;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import jakarta.persistence.Table;

@Entity
@Table(name = "quiz_answer")
public class QuizAnswer {
	
	@EmbeddedId
	private QuizAnswerId id;
	
	@CreationTimestamp
	@Column(name="created_at")
	private LocalDateTime createdAt;
	
	@ManyToOne
	@JoinColumn(name = "quiz_question_id") // DB column name
	@MapsId(value = "quizQuestionId")     // Field in ID class
	private QuizQuestion quizQuestion;

	@ManyToOne
	@JoinColumn(name = "user_id") // DB column
	@MapsId(value = "userId")     // Field in ID class
	private User user;
	
	@ManyToOne
	@JoinColumn(name = "choice_id")
	private Choice choice;
	
	public QuizAnswer() {
		super();
	}
	
	public QuizAnswerId getId() {
		return id;
	}

	public void setId(QuizAnswerId id) {
		this.id = id;
	}



	public LocalDateTime getCreatedAt() {
		return createdAt;
	}



	public void setCreatedAt(LocalDateTime createdAt) {
		this.createdAt = createdAt;
	}



	public QuizQuestion getQuizQuestion() {
		return quizQuestion;
	}



	public void setQuizQuestion(QuizQuestion quizQuestion) {
		this.quizQuestion = quizQuestion;
	}



	public User getUser() {
		return user;
	}



	public void setUser(User user) {
		this.user = user;
	}



	public Choice getChoice() {
		return choice;
	}



	public void setChoice(Choice choice) {
		this.choice = choice;
	}



	@Override
	public String toString() {
		return "QuizAnswer [createdAt=" + createdAt + "]";
	}

	
}

File: ngLessonLocker/src/app/models/quiz-answer.ts
import { Choice } from './choice';
import { User } from './user';

class QuizAnswerId {
  userId: number;
  quizQuestionId: number;

  constructor(userId: number = 0, quizQuestionId: number = 0) {
    this.userId = userId;
    this.quizQuestionId = quizQuestionId;
  }
}

export class QuizAnswer {
  // userId: number;
  id: QuizAnswerId | null = null;
  quizQuestionId: number;
  createdAt: Date | null;
  choiceId: number;
  // user: User;
  choice: Choice;

  constructor(
    // userId: number = 0,
    id: QuizAnswerId,
    quizQuestionId: number = 0,
    createdAt: Date | null = null,
    choiceId: number = 0,
    // user: User,
    choice: Choice
  ) {
    // this.userId = userId;
    this.id = id;
    this.quizQuestionId = quizQuestionId;
    this.createdAt = createdAt;
    this.choiceId = choiceId;
    // this.user = user;
    this.choice = choice;
  }
}
